cadl-tests/cadl-project.yaml

```diff
@@ -1,2 +1,4 @@
 emitters:
-  '@azure-tools/cadl-java': true
+  '@azure-tools/cadl-java':
+    generate-models: true
```

Comment:
The settings will directly go into `CadlPlugin`'s `SETTINGS_MAP`, thus readable by `JavaSettings`.

Reply:
we probably don't want to expose `generate-models` yet (not expect user to use it).